penguin/tools/providers/__init__.py

```
@@ -0,0 +1,13 @@
+"""Tool provider adapters for dynamic external tool surfaces."""
+
+from __future__ import annotations
+
+__all__ = ["MCPToolProvider"]
+
+
+def __getattr__(name: str):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `__getattr__`

(ANN202)